util/ranger/ranger.go

```diff
@@ -308,30 +315,45 @@ func points2TableRanges(sctx sessionctx.Context, rangePoints []*point, tp *types
 			HighExclude: endPoint.excl,
 			Collators:   []collate.Collator{collate.GetCollator(tp.GetCollate())},
 		}
+		if len(ranges) == 0 && rangeMaxSize > 0 && ran.MemUsage()*int64(len(rangePoints))/2 > rangeMaxSize {
+			return FullRange(), true, nil
+		}
 		ranges = append(ranges, ran)
 	}
-	return ranges, nil
+	return ranges, false, nil
 }
 
 // buildColumnRange builds range from CNF conditions.
-func buildColumnRange(accessConditions []expression.Expression, sctx sessionctx.Context, tp *types.FieldType, tableRange bool, colLen int) (ranges []*Range, err error) {
+// rangeMaxSize is the max memory limit for ranges. O indicates no memory limit.
```

Comment:
After all of the `todo` have been finished. Can we remove this parameter. Just use the `ctx.GetSessionVars().RangeMaxSize` to replace.